src/java.base/share/classes/java/lang/annotation/ElementType.java

```diff
@@ -26,106 +26,213 @@
 package java.lang.annotation;
 
 /**
- * The constants of this enumerated class provide a simple classification of the
- * syntactic locations where annotations may appear in a Java program. These
- * constants are used in {@link java.lang.annotation.Target Target}
- * meta-annotations to specify where it is legal to write annotations of a
- * given type.
+ * A syntactic location where an annotation may appear in Java code.
+ * An annotation interface may optionally restrict its usage to a
+ * particular subset of these locations using the {@link
+ * java.lang.annotation.Target @Target} meta-annotation.
  *
- * <p>The syntactic locations where annotations may appear are split into
- * <em>declaration contexts</em>, where annotations apply to declarations, and
- * <em>type contexts</em>, where annotations apply to types used in
- * declarations and expressions.
+ * <p>For example, an annotation of the following type may only appear
+ * as part of a type parameter or local variable declaration:
  *
- * <p>The constants {@link #ANNOTATION_TYPE}, {@link #CONSTRUCTOR}, {@link
- * #FIELD}, {@link #LOCAL_VARIABLE}, {@link #METHOD}, {@link #PACKAGE}, {@link
- * #MODULE}, {@link #PARAMETER}, {@link #TYPE}, and {@link #TYPE_PARAMETER}
- * correspond to the declaration contexts in JLS {@jls 9.6.4.1}.
+ * {@snippet id='example' :
+ * @Target({ElementType.TYPE_PARAMETER, ElementType.LOCAL_VARIABLE})
+ * public @interface MyAnnotation {}
+ * }
  *
- * <p>For example, an annotation whose interface is meta-annotated with
- * {@code @Target(ElementType.FIELD)} may only be written as a modifier for a
- * field declaration.
+ * <h2>Declaration annotations and type-use annotations</h2>
  *
- * <p>The constant {@link #TYPE_USE} corresponds to the type contexts in JLS
- * {@jls 4.11}, as well as to two declaration contexts: class and interface
- * declarations (including annotation declarations) and type parameter
- * declarations.
+ * <p>Most annotations in Java code are <b>declaration
+ * annotations</b>, which act like modifiers of declarations (such as
+ * a field declaration). The constants of this class cover all ten
```

Comment:
"such as a field declaration" seems like a carry-over from the old example of `@Target(ElementType.FIELD)`, which is obsolete in the current text. I suggest to remove, or to present a concrete example like "acts like modifiers of decrations, such as the transitive modifier on a field declaration."

The asserted number "ten" might be outdated quickly and forgotten to be updated when we add new declarations such as pattern bindings.

Reply:
It's meant to help make this more concrete for readers who aren't accustomed to speaking abstractly about "declarations". I've actually expanded it to "field or method" now. I certainly don't want to reference any particular modifier though. Does it work now?

I do realize that mentioning counts of things of is a pretty dodgy idea, almost "daring" the text to go stale. I could go either way about it but I want to try to defend it for a moment though.

The first question is whether citing a number like this has any positive value in the first place. This is absolutely subjective and it could just be my own "psychology" speaking only, but I think it lends "concreteness". When I read things like this it gives me more of a feeling of security that I'm really getting the full story and I can count on it. I don't know, I can't claim any more objective benefit than that, really.

As for the staleness issue, I'd claim that avoiding the numbers would basically mean we're hoping that the text might stay "accidentally current", when really we need to re-review the full text any time anything changes to quite *actively* search for what to freshen up. If a stale number helped us notice that the text hadn't be rethought yet, maybe it would be a good thing...... meh?

Reply:
Yep, the "field or method declaration" is much clearer. I felt the old text "field declaration" was trying to emphasize the modifier status of annotations.

And for count: that's fair. We bump Java version every six months, too!